Framework/Geometry/src/Objects/Object.cpp

```diff
@@ -1833,17 +1834,11 @@ int Object::searchForObject(Kernel::V3D &point) const {
   Kernel::V3D testPt;
   if (isValid(point))
     return 1;
-  std::vector<Kernel::V3D> axes;
-  axes.reserve(6);
-  axes.push_back(Kernel::V3D(1, 0, 0));
-  axes.push_back(Kernel::V3D(-1, 0, 0));
-  axes.push_back(Kernel::V3D(0, 1, 0));
-  axes.push_back(Kernel::V3D(0, -1, 0));
-  axes.push_back(Kernel::V3D(0, 0, 1));
-  axes.push_back(Kernel::V3D(0, 0, -1));
-  std::vector<Kernel::V3D>::const_iterator dir;
-  for (dir = axes.begin(); dir != axes.end(); ++dir) {
-    Geometry::Track tr(point, (*dir));
+  std::array<Kernel::V3D, 6> axes = {{V3D(1., 0., 0.), V3D(-1., 0., 0.),
+                                      V3D(0., 1., 0.), V3D(0., -1., 0.),
+                                      V3D(0., 0., 1.), V3D(0., 0., -1.)}};
+  for (const auto &dir : axes) {
```

Comment:
Cool trick I learned only recenty, you can use range-based for with an initializer list! Something like this:

```cpp
for (const auto &dir : {{V3D(1., 0., 0.), V3D(-1., 0., 0.), V3D(0., 1., 0.),
                         V3D(0., -1., 0.), V3D(0., 0., 1.), V3D(0., 0., -1.)}} ) {
```